jfrog-cli/artifactory/commands/config.go

```diff
@@ -145,20 +137,44 @@ func getConfigurationFromUser(details, defaultDetails *config.ArtifactoryDetails
 	return nil
 }
 
-func readSshKeyPathFromConsole(details, savedDetails *config.ArtifactoryDetails) error {
+func getSshKeyPath(details *config.ArtifactoryDetails) error {
+	// If path not provided as a key, read from console:
 	if details.SshKeyPath == "" {
-		ioutils.ScanFromConsole("SSH key file path", &details.SshKeyPath, savedDetails.SshKeyPath)
+		ioutils.ScanFromConsole("SSH key file path (optional)", &details.SshKeyPath, "")
+	}
+
+	// If path still not provided, return and warn about relying on agent.
+	if details.SshKeyPath == "" {
+		log.Info("SSH Key path not provided. Specify a key path using flags, or rely on ssh-agent only.")
+		return nil
 	}
 
+	// If SSH key path provided, check if exists:
 	details.SshKeyPath = clientutils.ReplaceTildeWithUserHome(details.SshKeyPath)
 	exists, err := fileutils.IsFileExists(details.SshKeyPath, false)
 	if err != nil {
 		return err
 	}
-	if !exists {
-		log.Warn("Could not find SSH key file at:", details.SshKeyPath)
+
+	if exists {
+		sshKeyBytes, err := ioutil.ReadFile(details.SshKeyPath)
+		if err != nil {
+			return nil
+		}
+		encryptedKey, err := auth.IsEncrypted(sshKeyBytes)
+		// If exists and not encrypted (or error occurred), return without asking for passphrase
+		if err != nil || !encryptedKey {
+			return err
+		} else {
+			log.Info("The key file at the specified path is encrypted.")
+			log.Info("Please pass the ssh-passphrase as a flag when authentication required.")
+		}
+	} else {
+		log.Info("Could not find key in provided path. You may place the key file there later.")
+		log.Info("If you choose to use an encrypted key, you may pass the passphrase as a flag when authentication required.")
```

Comment:
1. We can merge the above to log info messages into one message,
2. I suggest rephrasing:
"...you may pass the passphrase as a flag when authentication required."
to:
"...you may pass the passphrase as an option with every command."

The above two comments also applies to the other 2 log messages in this function.